### PR TITLE
feat(auth): Add ResourcesNameable to BasicAmazonDeployDescription, default to no authorization

### PR DESCRIPTION
**Pull request by jonsie**
Additionally, corrected some terminology (authn -> authz) and removed a dangerous `requiresAuthentication` boolean from `DescriptionAuthorizer` that could potentially cause authz to skip on resources.

Authorizing these resources is no doubt backwards incompatible, and not something we can enable at the moment.  However, I would like to collect some data internally to see how often we deploy server groups with resources that fall outside the context of the application frigga name -- that will help inform us how we can go about rolling out a change that authorizes application resources.

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java [hunk 1]
```diff
@@ -74,19 +76,25 @@ public void authorize(T description, Errors errors) {
...
     List<String> applications = new ArrayList<>();
     boolean requiresApplicationRestriction = true;
-    boolean requiresAuthentication = true;
 
     if (description instanceof AccountNameable) {
       AccountNameable accountNameable = (AccountNameable) description;
-      account = accountNameable.getAccount();
 
       requiresApplicationRestriction = accountNameable.requiresApplicationRestriction();
-      requiresAuthentication = accountNameable.requiresAuthentication(opsSecurityConfigProps);
```
Comment (jonsie) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java]: This boolean had to go -- I don't want someone to use it later in the code and then accidentally skip a resource that requires authorization.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/BasicAmazonDeployAtomicOperationConverter.groovy [hunk 2]
```diff
@@ -17,23 +17,46 @@
...
   AtomicOperation convertOperation(Map input) {
     new DeployAtomicOperation(convertDescription(input))
   }
 
   BasicAmazonDeployDescription convertDescription(Map input) {
     def converted = objectMapper.convertValue(input, BasicAmazonDeployDescription)
     converted.credentials = getCredentialsObject(input.credentials as String)
-    converted
+
+    if (converted.securityGroups) {
```
Comment (robzienert) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/BasicAmazonDeployAtomicOperationConverter.groovy]: I'm not a big fan of groovy truth - it makes things more difficult to reason about when converting things to Java.

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ResourcesNameable.java [hunk 3]
```diff
@@ -36,4 +36,8 @@
         .map(name -> Names.parseName(name).getApp())
         .collect(Collectors.toList());
   }
+
+  default boolean requiresAuthorization() {
```
Comment (robzienert) [inline clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ResourcesNameable.java]: Docs needed on interfaces and their methods.

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java [hunk 4]
```diff
@@ -60,6 +61,7 @@ public DescriptionAuthorizer(
     this.fiatPermissionEvaluator = fiatPermissionEvaluator.orElse(null);
     this.opsSecurityConfigProps = opsSecurityConfigProps;
 
+    this.skipAuthorizationId = registry.createId("authorization.skip");
```
Comment (robzienert) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java]: Not a very descriptive metric name - skip what?

Comment (jonsie) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java]: Hm, I can change this to the past tense `authorization.skipped` - that seems more descriptive to me (basically, this metric is saying that we skipped authorization on a resource).

Comment (robzienert) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/deploy/DescriptionAuthorizer.java]: That sounds good.

**Comment by robzienert** [comment]
Ping.

**Comment by jonsie** [comment]
Decided to get this over the line, should be ready to go.

**Pull request merged**